server/container_list.go

```diff
@@ -35,28 +37,29 @@ func (s *Server) filterContainerList(ctx context.Context, filter *pb.ContainerFi
 		if err != nil {
 			// If we don't find a container ID with a filter, it should not
 			// be considered an error.  Log a warning and return an empty struct
-			log.Warnf(ctx, "unable to find container ID %s", filter.Id)
-			return []*oci.Container{}
+			log.Warnf(ctx, "Unable to find container ID %s", filter.Id)
+			return nil
 		}
 		c := s.ContainerServer.GetContainer(id)
 		if c != nil {
 			switch {
 			case filter.PodSandboxId == "":
 				return []*oci.Container{c}
-			case c.Sandbox() == filter.PodSandboxId:
+			case strings.HasPrefix(c.Sandbox(), filter.PodSandboxId):
 				return []*oci.Container{c}
 			default:
-				return []*oci.Container{}
+				return nil
 			}
 		}
 	} else if filter.PodSandboxId != "" {
-		pod := s.ContainerServer.GetSandbox(filter.PodSandboxId)
-		if pod == nil {
-			return []*oci.Container{}
+		for _, sb := range s.ListSandboxes() {
+			if strings.HasPrefix(sb.ID(), filter.PodSandboxId) {
+				return sb.Containers().List()
```

Comment:
hmm. Typically in containerd the short id is only expanded/matched if there is but one match. This code is iterating through the list and taking just the first match. Is that typical for short id processing in crio?